src/App.js

```diff
@@ -3,17 +3,56 @@ import GridLayout from "./components/GridLayout";
 import FullLayout from "./components/FullLayout";
 import { connect } from "react-redux";
 import { Route, withRouter } from "react-router-dom";
+import { getConfiguredNow } from "./reducers";
 
-import { fetchAlarms } from "./actions";
+import { fetchAlarms, setDateTimeAction } from "./actions";
 import styles from "./App.css";
 
 class App extends Component {
+  componentWillMount() {
+    this.setDateTimeActionWithNow();
+    const that = this;
+    // update redux time every minute (60000 miliseconds) because redux only saves time on the minute accurate
+    setInterval(function() {
+      that.setDateTimeActionWithNow();
+    }, 60000);
+  }
+
   componentDidMount() {
     if (!this.props.iframeModeActive) {
       this.props.fetchAlarms();
     }
   }
 
+  setDateTimeActionWithNow() {
+    const jsDateObject = new Date();
+    const utcMinutes = jsDateObject.getUTCMinutes();
+    const utcMinutesZeroPrefixed =
+      utcMinutes < 10 ? "0" + utcMinutes : utcMinutes;
+
+    const dateStr =
+      jsDateObject.getUTCFullYear() +
+      "-" +
+      (jsDateObject.getUTCMonth() + 1) +
+      "-" +
+      jsDateObject.getUTCDate();
+    const timeStr = jsDateObject.getUTCHours() + ":" + utcMinutesZeroPrefixed;
+
+    this.props.setDateTimeAction(dateStr, timeStr);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      !prevProps.configuredNow ||
+      (prevProps.configuredNow && !this.props.configuredNow) ||
+      this.props.configuredNow.getTime() != prevProps.configuredNow.getTime()
+    ) {
+      if (!this.props.iframeModeActive) {
+        this.props.fetchAlarms();
+      }
```

Comment:
Er moet een functie komen op de Redux store, die "now" oplevert; en dat is ofwel configuredNow als die er is, of de laatst geupdate setDateTime.

Reply:
Ja klopt. Deze bestaat ook namelijk de functie " getConfiguredNow(state)" zoals die in de reducer.js staat. Deze gebruiken we hier ook, namelijk zie de functie mapStateToProps hieronder. De functie hierboven bepaald dus niet welke 'now' gebruikt wordt, maar vergelijkt de huidige now met de voorgaande now en bepaald zo of een nieuwe render cycle nodig is. De naamgeving is misschien wel verwarrend.